test/e2e/multihoming_utils.go

```diff
@@ -415,3 +417,53 @@ func reachToServerPodFromClient(cs clientset.Interface, serverConfig podConfigur
 
 	return fmt.Errorf("pod not running. /me is sad")
 }
+
+func isPortInUse(host, port string) bool {
+	conn, err := net.Dial("tcp", net.JoinHostPort(host, port))
+	if err != nil {
+		return false
+	}
+	conn.Close()
+	return true
+}
+
+func findInterfaceByIP(targetIP string) (string, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return "", err
+		}
+
+		for _, addr := range addrs {
+			ip, _, err := net.ParseCIDR(addr.String())
+			if err != nil {
+				return "", err
+			}
+
+			if ip.String() == targetIP {
+				return iface.Name, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Interface with IP %s not found", targetIP)
+}
+
+func getNetworkGateway(cli *client.Client, networkName string) (string, error) {
```

Comment:
maybe you're trying to do too much; while this works (now) it relies on implementation details - like IPAM on the CRI network. If the runtime stops configuring IPAM on the secondary network, this will stop working.

Maybe we should do less and just be passed the name of the bridge implementing the secondary network via env variable, just like it is being done for the other tests ? https://github.com/ovn-org/ovn-kubernetes/blob/0beaf4ee34981a65a888abf6bfca587a7a5e5671/test/e2e/external_gateways.go#L44

@jcaamano thoughts ?

Reply:
imho the CRI isn't going to change so fast, and only if it does we can fix it
meanwhile it will allow less parameters to be set, and auto detect with native go code

Reply:
I understand the motivation, and the advantages, they're clear.

Let's hear from ovn-k maintainers.